### ISSUE TITLE
[Bug] can clash.meta transfer IPv6 traffic?

### ISSUE THREAD

**Issue by Azadzadeh** [MetaCubeX/mihomo#794]
### Verify steps

- [X]  确保你使用的是**本仓库**最新的的 clash 或 clash Alpha 版本 Ensure you are using the latest version of Clash or Clash Premium from **this repository**. 
- [X]  如果你可以自己 debug 并解决的话，提交 PR 吧 Is this something you can **debug and fix**? Send a pull request! Bug fixes and documentation fixes are welcome. 
- [X]  我已经在 [Issue Tracker](……/) 中找过我要提出的问题 I have searched on the [issue tracker](……/) for a related issue. 
- [X]  我已经使用 Alpha 分支版本测试过，问题依旧存在 I have tested using the dev branch, and the issue still exists. 
- [X]  我已经仔细看过 [Documentation](https://wiki.metacubex.one/) 并无法自行解决问题 I have read the [documentation](https://wiki.metacubex.one/) and was unable to solve the issue. 
- [X]  这是 Clash 核心的问题，并非我所使用的 Clash 衍生版本（如 OpenClash、KoolClash 等）的特定问题 This is an issue of the Clash core *per se*, not to the derivatives of Clash, like OpenClash or KoolClash. 

### Clash version

Clash Meta v1.16.0

### What OS are you seeing the problem on?

Linux

### Clash config

```yaml
port: 8080
socks-port: 1080 

mode: rule
log-level: debug
ipv6: true

dns:
  enable: true
  listen: 0.0.0.0:53
  ipv6: true

tun:
  enable: true
  stack: system
  dns-hijack:
    - any:53
  auto-route: true
  auto-detect-interface: true
```

### Description

does clash.meta have ability to pass ipv6 traffic?
my proxy server is dual-stack. inside it, i can ping6 an ipv6 or ssh to a ipv6 address.
but, at client side, network is only ipv4.

I thought if i activate a proxy to that server in TUN mode , i'd be able to ping6 at client side but this doesn't work:
```
$ ping 2001:4860:4860::8888
connect: Network is unreachable
```
no trace about this query is shown on the logs. 
i do `netstat -lntup`, no tcp6 or udp6 connections are listening on 1080 or 8080.

However, `nslookup ipv6.google.com` works and i see a line in logs related to it.

In short, i want to ping6 or `ssh -6` at an ipv4-only client side using the dual-stack proxy server.

**Comment by xishang0128** [comment]
@Azadzadeh Pure ipv6 domain name/ip should be used for non-ICMP testing. Clash will not proxy ICMP.

**Comment by xishang0128** [comment]
@Azadzadeh Does your node server provide aaaa resolve and ipv6 outbound?

**Comment by closehandle** [comment]
You need to configure Fake IP mode for DNS. As far as I know, it doesn't do any additional configuration for IPv6 routing. What this means is that it will not handle IPv6 traffic on TUN. If you use the regular HTTP/Socks5 proxy you'll find it works fine!

**Comment by xishang0128** [comment]
@closehandle The tun of clash.meta handles and routes ipv6 traffic and does not need to be set to fakeip

**Comment by xishang0128** [comment]
@Azadzadeh You can try setting dns.enhaced-mode to redir-host

**Comment by xishang0128** [comment]
@Azadzadeh Use ip a or ifconfig to check whether the tun device has an ipv6 address

**Comment by xishang0128** [comment]
@Azadzadeh Then tun should receive ipv6 traffic

**Comment by lux5am** [comment]
try tun gvisor

**Comment by xishang0128** [comment]
> @xishang0128 `sudo tcpdump -i Meta`当我运行该curl命令时，仅显示DNS活动（AAAA和NXDomain）。

Sorry, my test is normal

![image](https://github.com/MetaCubeX/Clash.Meta/assets/95160953/c0b3148e-21df-4ca0-81b7-82df8cdf3bb2)
![image](https://github.com/MetaCubeX/Clash.Meta/assets/95160953/439c085f-fa5e-47d3-9cc3-4686fce40caf)
![image](https://github.com/MetaCubeX/Clash.Meta/assets/95160953/e1f18219-de3f-4115-9cc7-9b85e75024fb)

**Comment by xishang0128** [comment]
> @xishang0128 `sudo tcpdump -i Meta`当我运行该curl命令时，仅显示DNS活动（AAAA和NXDomain）。

Maybe you should try fakeip to test whether dns hijack works.

**Comment by Skyxim** [comment]
@Azadzadeh If you use REDIR-HOST, you must ensure that there is IPv6 locally, otherwise the proxy application cannot initiate a request.

**Comment by Skyxim** [comment]
@Azadzadeh In your case, it should be the local computer, specifically it should be applied to meta by the proxy.

**Comment by Skyxim** [comment]
@Azadzadeh fe80:: just like 169.254.x.x of IPv4, it will not be used in practice. It is used for host communication.

**Comment by closehandle** [comment]
The machine on which you deployed the TUN does not support access to an IPv6 network, so there is no IPv6 default route, and Clash Meta has not configured an IPv6 address and route for the TUN device, so it rejected your connection at the routing decision stage

That's why you'll get an error if you try to access the IPv6 address directly. The routing decision takes precedence over iptables, and the connection will be rejected before it is processed by iptables!

Fake IP only provides A record resolution, so it works, but you'll still fail when accessing IPv6 addresses directly

You can configure a default route yourself so that the kernel doesn't reject your connection during the routing decision phase
```
ip -6 route add default dev tun0
```

> @xishang0128 Thanks for testing this and taking screenshots ❤️ I found one of my problems when i restarted my computer. when i changed clash configs and restarted it, somehow some parameters in my network interface where not getting updated (some kind of cache?) . so after a config change, i also turned off and on my NetworkManager. Now it works in TUN mode for this:
> 
> ```yaml
> mode: rule
> log-level: debug
> ipv6: true
> 
> tun:
>   enable: true
>   stack: system
>   dns-hijack:
>     - any:53
>   auto-route: true
>   auto-detect-interface: true
> 
> dns:
>   enable: true
>   ipv6: true
>   enhanced-mode: fake-ip
> ```
> 
> This works now: `curl ipv6.google.com` Also, it looks like general `ipv6: true` and `dns.ipv6:true` are irrelevant for this to work. however, `enhanced-mode` must be set to fake-ip. otherwise it won't work.
> 
>     * Is there a reason that fake-ip is necessary for this? it doesn't return true IPs in this mode so it would be a bit problematic for development
> 
>     * Also, unfortunately, it still can't connect to direct IPv6 addresses:
>       `$ curl -vvv ipv6.google.com` => works
>       `$ curl -x socks5://localhost:1080 -vvv http://[2001:4860:4860::8888]/` => works
>       `$ curl -vvv http://[2001:4860:4860::8888]/` => `Immediate connect fail for 2001:4860:4860::8888: Network is unreachable`
>       This was for above config and after restarting my network card.

**Comment by stevejohnson7** [comment]
This problem seems to be solved. Can I close it? or change to a discussion?
